typescript/sdk/src/deploy/HyperlaneDeployer.ts

```diff
@@ -677,4 +689,23 @@ export abstract class HyperlaneDeployer<
 
     return receipts.filter((x) => !!x) as ethers.ContractReceipt[];
   }
+
+  protected async resolveInterchainAccountAsOwner(
+    chain: ChainName,
+    owner: Owner,
+  ): Promise<Address> {
+    if (typeof owner === 'string') {
+      return owner;
+    } else {
+      const routerAddress = this.cachedAddresses[chain].interchainAccountRouter;
+      if (!routerAddress) {
+        throw new Error('InterchainAccountRouter not deployed');
+      }
+      const router = InterchainAccount.fromAddressesMap(
+        this.cachedAddresses,
+        this.multiProvider,
+      );
+      return router.deployAccount(chain, owner);
```

Comment:
yaa that's an option but if we're gonna set the address as owner, I don't see the downside in deploying.

Reply:
the downside as I mentioned elsewhere is submitting additional transactions unnecessarily

Reply:
I make the `getLocalInterchainAccount` before `getDeployedInterchainAccount` now so this means we won't be submitting transactions unnecessarily.

Reply:
I dont understand why you are adamant about this
imo its a nice feature that we only bear the cost of deploy once an ICA call is made and can otherwise derive counterfactually but I digress
fwiw this also requires submitting a TX on the destination chain which is what we are trying to avoid with this feature